seismicpro/survey/metrics.py

```diff
@@ -29,3 +40,686 @@ def plot(self, ax, coords, index, sort_by=None, **kwargs):
+    amplitude value.
+
+    The metric is highly depends on muter, if muter isn't strong enough, the metric will overreact to the first breaks.
+
+    Parameters
+    ----------
+    muter : Muter
+        A muter to use.
+    name : str, optional, defaults to "spikes"
+        Metrics name.
+
+    Attributes
+    ----------
+    ?? Do we want to describe them ??
+    """
+    name = "spikes"
+    min_value = 0
+    max_value = None
+    is_lower_better = True
+    threshold = 2
+
+    @property
+    def description(self):
+        """String description of tracewise metric"""
+        return "Traces with spikes"
+
+    @staticmethod
+    @njit(nogil=True)
+    def numba_get_mask(traces):
+        """QC indicator implementation.
+
+        The resulted 2d mask shows the deviation of the ampluteds of an input gather.
+        """
+        res = np.zeros_like(traces)
+        for i in range(traces.shape[0]):
+            nan_indices = np.nonzero(np.isnan(traces[i]))[0]
+            if len(nan_indices) > 0:
+                j = nan_indices[-1] + 1
+                if j < traces.shape[1]:
+                    traces[i, :j] = traces[i, j]
+                else:
+                    traces[i, :] = 0
+            res[i, 1: -1] = np.abs(traces[i, 2:] + traces[i, :-2] - 2*traces[i, 1:-1]) / 3
+        return res
+
+
+class Autocorrelation(MuteTracewiseMetric):
+    """Trace correlation with itself shifted by 1.
+
+    The metric is highly depends on muter, if muter isn't strong enough, the metric will overreact to the first breaks.
+
+    Parameters
+    ----------
+    muter : Muter
+        A muter to use.
+    name : str, optional, defaults to "autocorrelation"
+        Metrics name.
+    """
+    name = "autocorrelation"
+    min_value = -1
+    max_value = 1
+    is_lower_better = False
+    threshold = 0.8
+
+    @property
+    def description(self):
+        """String description of tracewise metric"""
+        return f"Traces with autocorrelation less than {self.threshold}"
+
+    @staticmethod
+    @njit(nogil=True)
+    def numba_get_mask(traces):
+        """QC indicator implementation."""
+        # TODO: descide what to do with almost nan traces (in 98% in trace are nan, it almost always will have -1 val)
+        res = np.empty_like(traces[:, 0])
+        for i in range(traces.shape[0]):
+            res[i] = np.nanmean(traces[i, 1:] * traces[i, :-1])
+        return res
+
+class TraceAbsMean(TracewiseMetric):
+    """Absolute value of the trace's mean scaled by trace's std.
+
+    Parameters
+    ----------
+    name : str, optional, defaults to "trace_absmean"
+        Metrics name.
+    """
+    name = "trace_absmean"
+    is_lower_better = True
+    threshold = 0.1
+
+    @property
+    def description(self):
+        """String description of tracewise metric"""
+        return f"Traces with inverce CV bigger than {self.threshold}"
```

Comment:
```suggestion
        return f"Traces with inverse CV greater than {self.threshold}"
```
But I'd call it "mean divided by std" like in the next metric to be more clear

Reply:
Done